perf/perf.yaml

```diff
@@ -0,0 +1,83 @@
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+apiVersion: apps/v1
+kind: DaemonSet
+metadata:
+  name: enable-perf
+  labels:
+    app: enable-perf
+spec:
+  selector:
+    matchLabels:
+      name: enable-perf
+  template:
+    metadata:
+      labels:
+        name: enable-perf
+    spec:
+      nodeSelector:
+        "enable-perf": "true"
+      hostPID: true
+      volumes:
+        - name: host
+          hostPath:
+            path: /
+      containers:
+        - name: enable-perf
+          image: debian
+          imagePullPolicy: Always
+          volumeMounts:
+            - name: host
+              mountPath: /host
+          securityContext:
+            privileged: true
+          env:
+            # TARGET_PGREP options
+            # Values:
+            #   empty or unset to do full trace
+            #   pgrep query to filter
+            - name: TARGET_PGREP
+              value: "kubelet"
+          command:
+            - /bin/bash
+            - -c
+            - |
+
+              set -o errexit
+              set -o pipefail
+              set -o nounset
+
+              MAX_LOG_SIZE="16777215" # 16 MB
+              MAX_LOGS="2000"
+
+              apt-get update && apt-get install -y linux-perf procps daemontools
+              PERF="/usr/bin/perf_4.19"
```

Comment:
use just `perf`? why pin to 4.19?

Reply:
There's some weirdness with how perf gets installed and `perf` itself is not usable:

```
root@enable-perf-78rc7:/# perf
/usr/bin/perf: line 13: exec: perf_5.0: not found
E: linux-perf-5.0 is not installed.
root@enable-perf-78rc7:/#
```

it does install perf in `/usr/bin/perf_4.19`, which works fine, so used that.

Reply:
I ended up changing this to just compile from source since it was alot simpler than getting the right package for the kernel version.